honssh/base_auth_handler.py

```diff
@@ -35,29 +35,33 @@
 
 class Base_Auth():
     
-    def __init__(self, server):
+    def __init__(self, server, name):
         self.server = server
+        self.name = name
         self.auth_plugin = None
         self.cfg = self.server.cfg
-        
+
+        self.connection_timeout = 10
```

Comment:
I'll wait for this merge before adding the new pull requests. As it may create some conflicts.